yt/yt/client/table_client/composite_compare.cpp

```diff
@@ -276,4 +308,112 @@ TFingerprint CompositeFarmHash(TYsonStringBuf value)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+std::optional<TYsonString> TruncateCompositeValue(TYsonStringBuf value, i64 size)
+{
+    YT_ASSERT(value.GetType() == EYsonType::Node);
+
+    if (size <= 0) {
```

Comment:
Is negative value possible here?

Reply:
Not at the moment. Maybe this should be a verify instead.

Reply:
Now it could be a zero, so let's keep this if.